### PR TITLE
sclang: changed some boost code to std

### PR DESCRIPTION
**Pull request by llloret**
**This is to have windows MSVC compile on master again.**

Because MSVC does not support some C++11 constructs that boost is using.

Note that we can probably streamline the code with some MACROs or
something, but I want to make it very explicit for now, to help with the
PR discussion.

### CODE REVIEW

**Comment by timblechmann** [comment]
thanks for looking into this! it looks pretty good, just two comments:

* i wouldn't use macros to 'simplify' the code, as executors (used in the `boost::async` codepath) are essentially coming from a c++17 proposal. so having it explicit is good, one can remove the platform-specifics once c++17 is out.

* there is one potential problem with `std::launch::async`: it will create one thread per `future`, which might create a significant overhead (creating threads). the main point for using an executor-based approach is to use only as many threads as there are CPUs.

File: lang/LangSource/PyrObject.cpp [hunk 1]
```diff
@@ -1211,33 +1222,53 @@ void buildBigMethodMatrix()
...
 		auto future = boost::async( pool, calcRowStats, bigTable, filledSelectors, numClasses, numSelectors, beginSelectorIndex, endSelectorIndex );
+#endif
 		columnDescriptorsWithStats.push_back( std::move(future) );
 	}
 
 	calcRowStats(bigTable, filledSelectors, numClasses, numSelectors, 0, std::min( selectorsPerJob, numSelectors) );
 
 	for( auto & future : columnDescriptorsWithStats ) {
+#ifdef _MSC_VER
+        future.wait();
```
Comment (timblechmann) [inline lang/LangSource/PyrObject.cpp]: please try not to mix tabs and spaces.

Comment (llloret) [inline lang/LangSource/PyrObject.cpp]: Ok, I will change the async to deferred.

Sorry about the space, I will fix that too, of course.

> >  		columnDescriptorsWithStats.push_back( std::move(future) );
> >  	}
> >
> >  	calcRowStats(bigTable, filledSelectors, numClasses, numSelectors, 0, std::min( selectorsPerJob, numSelectors) );
> >
> >  	for( auto & future : columnDescriptorsWithStats ) {
> > +#ifdef _MSC_
>
>
>

**Comment by bagong** [comment]
Master now builds fine with VS 2013 with both boost 1.60 and 1.61. It think maybe after Tim gave a lgtm this should go in as quickly as possible.

**Comment by bagong** [comment]
Windows fully back on master, yey! Thanks @llloret !

**Pull request merged**